graphql_compiler/compiler/helpers.py

```diff
@@ -41,6 +54,24 @@ def get_field_type_from_schema(schema_type, field_name):
         return schema_type.fields[field_name].type
 
 
+def get_vertex_field_type(current_schema_type, vertex_field_name):
+    """Return the type of the vertex within the specified vertex field name of the given type."""
+    # According to the schema, the vertex field itself is of type GraphQLList, and this is
+    # what get_field_type_from_schema returns. We care about what the type *inside* the list is,
+    # i.e., the type on the other side of the edge (hence .of_type).
+    # Validation guarantees that the field must exist in the schema.
+    if not is_vertex_field_name(vertex_field_name):
+        raise AssertionError(u'Trying to load the vertex field type of a non-vertex field: '
+                             u'{} {}'.format(current_schema_type, vertex_field_name))
+
+    raw_field_type = get_field_type_from_schema(current_schema_type, vertex_field_name)
+    if not isinstance(strip_non_null_from_type(raw_field_type), GraphQLList):
```

Comment:
Why are we stripping _only_ Non-Null Types ? Is it possible for there to be a type : `[None, None]`, for example ?

Reply:
The GraphQL type system by default assumes that types are nullable. We don't currently use the non-null type annotations anywhere in the compiler, so we just blindly strip them everywhere.

Primer on GraphQL types
- `[String]`: nullable list of nullable strings
- `[String!]`: nullable list of non-nullable strings
- `[String]!`: non-nullable list of nullable strings
- `[String!]!`: non-nullable list of non-nullable strings